ci/docker/serve.build.Dockerfile

```diff
@@ -13,14 +14,17 @@ SHELL ["/bin/bash", "-ice"]
 
 COPY . .
 
-RUN pip install -U torch==1.9.0 torchvision==0.10.0
-RUN pip install -U -c python/requirements_compiled.txt \
-	tensorflow tensorflow-probability
+# Install custom Python version if requested.
+RUN if [[ -z $PYTHON_VERSION ]] ; then echo Not installing custom Python version ; else PYTHON=$PYTHON_VERSION ci/env/install-dependencies.sh ; fi
```

Comment:
I do not recommended doing this.

we should edit `oss-ci-base_build`, make it a matrix, so that it has different variations of python, rather than trying to overwrite the python version here.

the drawbacks of doing this way:
- now this image has two pythons, where it only needs one.
- this image becomes serve-specific, and cannot be used for other python 3.11 tests if other teams want to onboard more.

let me know if you need help.

Reply:
Agree in general. I'm just planning to do the minimal here thing to get postmerge tests running for serve against 3.11 and then @can-anyscale is planning to do some more general work to get python 3.11 builds working across the board